src/lib/Menu/Listener/ReorderByOrderNumberListener.php

```diff
@@ -9,63 +9,44 @@
 namespace EzSystems\EzPlatformAdminUi\Menu\Listener;
 
 use EzSystems\EzPlatformAdminUi\Menu\Event\ConfigureMenuEvent;
+use Knp\Menu\ItemInterface;
 
 final class ReorderByOrderNumberListener
 {
-    /**
-     * @param \EzSystems\EzPlatformAdminUi\Menu\Event\ConfigureMenuEvent $event
-     */
     public function reorderMenuItems(ConfigureMenuEvent $event): void
     {
         $menu = $event->getMenu();
-        $menuOrderArray = [];
-        $addLast = [];
-        $alreadyTaken = [];
 
-        foreach ($menu->getChildren() as $key => $menuItem) {
-            if ($menuItem->hasChildren()) {
-                $this->reorderMenuItems($menuItem);
-            }
-            $orderNumber = $menuItem->getExtra('orderNumber');
+        $this->recursiveReorderMenuItems($menu);
+    }
 
-            if ($orderNumber != null) {
-                if (!isset($menuOrderArray[$orderNumber])) {
-                    $menuOrderArray[$orderNumber] = $menuItem->getName();
-                } else {
-                    $alreadyTaken[$orderNumber] = $menuItem->getName();
-                }
-            } else {
-                $addLast[] = $menuItem->getName();
-            }
-        }
-        ksort($menuOrderArray);
+    private function recursiveReorderMenuItems(ItemInterface $menuItem): void
+    {
+        $menuItemList = [];
+        $unorderedMenuItemsList = [];
 
-        if (count($alreadyTaken)) {
-            foreach ($alreadyTaken as $key => $value) {
-                $keysArray = array_keys($menuOrderArray);
-                $position = array_search($key, $keysArray);
+        foreach ($menuItem->getChildren() as $key => $nestedMenuItem) {
+            if ($nestedMenuItem->hasChildren()) {
+                $this->recursiveReorderMenuItems($nestedMenuItem);
+            }
 
-                if ($position === false) {
-                    continue;
-                }
+            $orderNumber = $nestedMenuItem->getExtra('orderNumber');
 
-                $menuOrderArray = array_merge(
-                    array_slice($menuOrderArray, 0, $position),
-                    [$value],
-                    array_slice($menuOrderArray, $position)
-                );
+            if ($orderNumber === null) {
+                $unorderedMenuItemsList[] = $nestedMenuItem;
+            } else {
+                $menuItemList[$orderNumber][] = $nestedMenuItem;
```

Comment:
Doesn't it raise a warning? I'd check if `$menuItemList[$orderNumber]` is not set and initialize it as an array first.

Reply:
It's correct construction. You can dynamically allocate array "on demand" this way.

Reply:
It's just too PHP specific for my taste. Anyway I'm fine with this ✅